modules/pam_userdb/pam_userdb.c

```diff
@@ -213,32 +213,38 @@ user_lookup (pam_handle_t *pamh, const char *database, const char *cryptmode,
 	  } else if (ctrl & PAM_ICASE_ARG) {
 	    compare = -2;
 	  } else {
+	    /* libdb is not guaranteed to produce null terminated strings */
+	    char *pwhash = strndup(data.dptr, data.dsize);
+	    if (pwhash != NULL) {
```

Comment:
Please add an empty line after the declaration and also add compare = -2 otherwise if the strndup fails compare == 0 and that will signalize a password match! Perhaps also some syslog when the strndup fails might be appropriate.

Reply:
Thanks for spotting and your patience. I changed the code a bit to initialize `compare` to `-2` to be more safe against these kinds of overlooks. Does it make sense?

Reply:
I don't think `malloc` or any other memory allocation function is safe enough and they should be checked in case of failure. Take into account that pam is a fundamental software in Linux distributions and any unexplained failure can lead to hundreds of hours of work to reproduce and understand the failure.

Reply:
@ikerexxe There might be a misunderstanding. Of course the return value must be checked. We are talking about adding syslog message in case allocation fails.
Are we on the same page? Are you proposing that syslog message should be added for all the cases where allocation fails?

Reply:
> @ikerexxe There might be a misunderstanding. Of course the return value must be checked. We are talking about adding syslog message in case allocation fails.

Sorry, I had understood something different. 

> Are we on the same page? Are you proposing that syslog message should be added for all the cases where allocation fails?

If possible I'd add a syslog message for all allocations that fail.

Reply:
@ikerexxe ok, added syslog messages